src/test/java/org/candlepin/service/model/CloudCheckInEventTest.java

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCloudData;
+import org.candlepin.test.TestUtil;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+
+import java.util.Date;
+import java.util.List;
+
+public class CloudCheckInEventTest {
+
+    private static final String SYSTEM_UUID_KEY = "systemUuid";
+    private static final String CHECK_IN_KEY = "checkIn";
+    private static final String PROVIDER_KEY = "cloudProviderId";
+    private static final String ACCOUNT_KEY = "cloudAccountId";
+    private static final String CLOUD_OFFERINGS_KEY = "cloudOfferingIds";
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void testConstructorWithNullConsumerCloudData() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            new CloudCheckInEvent(null, mapper);
+        });
+    }
+
+    @Test
+    public void testConstuctorWithNullConsumer() {
```

Comment:
typo here, and in other methods in this class: `Constuctor` instead of `Constructor`

Reply:
updated throughout